arb-gateway/README.md

```diff
@@ -9,25 +9,24 @@ To get started, you need to have an RPC URL for both Ethereum Mainnet and Arbitr
 ## How to use arb-gateway locally via cloudflare dev env (aka wrangler)
 
 ```
-npm install -g bun
 cd arb-gateway
+npm install -g bun wrangler
 bun install
 touch .dev.vars
 ## set L1_PROVIDER_URL, L2_PROVIDER_URL, L2_ROLLUP
-yarn dev
+bun run dev
```

Comment:
I'm experiencing the installation and build errors and have decided to try https://github.com/ensdomains/evmgateway/pull/32 which includes these changes.

However, I'm still experiencing these errors. I'm not familiar with bun, but I see it's used as a monorepo package manager. 

I've noticed that popular repositories often use pnpm for their monorepo package manager. What do you think about switching from bun to pnpm for fixing the installation and build errors?

Reply:
What errors are you encountering if you follow the steps in the README?

Reply:
Cloning main branch of evmgateway

------------------------

## bun install

```
bun install v1.0.35 (940448d6)
40 |     "@eth-optimism/sdk/**/ethers": "^5.7.0"
         ^
warn: Bun currently does not support nested "resolutions"
   at /Users/chomtana/Desktop/projects/temp/evmgateway/package.json:40:5
  ⚙️  keccak [40/68] building l1-gateway...
src/index.ts(1,28): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.
src/L1ProofService.ts(8,52): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.
src/server.ts(3,28): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.
src/worker.ts(2,29): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.
src/worker.ts(12,38): error TS2307: Cannot find module '@ensdomains/evm-gateway' or its corresponding type declarations.

$ echo 'building l1-gateway...' && bun run clean && bun run build:cjs && bun run build:esm && bun run build:types
$ rm -fr _cjs _esm _types
$ tsc --project tsconfig.build.json --module commonjs --outDir ./_cjs --removeComments --verbatimModuleSyntax false && echo > ./_cjs/package.json '{"type":"commonjs"}'

error: prepare script from "@ensdomains/l1-gateway" exited with 2
```

------------------------

## bun run build

When building op-verifier (Randomly happen to other verifier too)

```
Running script "build" in workspace "op-verifier"
$ echo 'building op-verifier...' && bun hardhat compile
building op-verifier...
An unexpected error occurred:

TypeError: Cannot read properties of undefined (reading 'JsonRpcSigner')
    at Object.<anonymous> (/Users/chomtana/Desktop/projects/temp/evmgateway/node_modules/zksync-web3/build/src/signer.js:78:74)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Object.require.extensions.<computed> [as .js] (/Users/chomtana/Desktop/projects/temp/evmgateway/node_modules/ts-node/src/index.ts:1608:43)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (/Users/chomtana/Desktop/projects/temp/evmgateway/node_modules/zksync-web3/build/src/utils.js:20:18)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
error: "hardhat" exited with code 1
error: script "build" exited with code 1
The script failed 1
error: 1
error: script "workspace" exited with code 1
error: script "build" exited with code 1
```

Hot fix seems to be commenting these hardhat-deploy imports before running build script.

```
import 'hardhat-deploy';
import 'hardhat-deploy-ethers';
```

------------------------

## clones-with-immutable-args

If we have fixed above errors and remove bun.lockb, we will face with a new error on clones-with-immutable-args.

Bun can't resolve [Arachnid/clones-with-immutable-args#feature/create2](https://github.com/Arachnid/clones-with-immutable-args). It can only resolve the primary branch of the repo.

Even though it can resolve, it is not in the correct commit. So, I have to clone it to https://github.com/Chomtana/clones-with-immutable-args-for-evmgateway

Reply:
That feature has actually been merged into the parent repo: https://github.com/wighawag/clones-with-immutable-args/pull/19

So you can simply change to using the upstream version.

Reply:
The problem is this contract is using `predictAddress` function that has been renamed to `addressOfClone2 `

https://github.com/ensdomains/ens-contracts/blob/delegatable-resolver-with-factory/contracts/resolvers/DelegatableResolverFactory.sol

Reply:
I have opened a PR to fix this. We need to fix on ens-contracts first https://github.com/ensdomains/ens-contracts/pull/344